src/handlers/payout/post.ts

```diff
@@ -121,6 +130,119 @@ export const incentivizeComments = async () => {
+  );
+  if (permitComments.length > 0) {
+    logger.info(`incentivizePullRequestReviews: skip to generate a permit url because it has been already posted`);
+    return;
+  }
+
+  const assignees = issue?.assignees ?? [];
+  const assignee = assignees.length > 0 ? assignees[0] : undefined;
+  if (!assignee) {
+    logger.info("incentivizePullRequestReviews: skipping payment permit generation because `assignee` is `undefined`.");
+    return;
+  }
+
+  const prReviews = await getAllPullRequestReviews(context, linkedPullRequest.number, "full");
+  logger.info(`Getting the PR reviews done. comments: ${JSON.stringify(prReviews)}`);
+  const prReviewsByUser: Record<string, string[]> = {};
+  for (const review of prReviews) {
+    const user = review.user;
+    if (!user) continue;
+    if (user.type == UserType.Bot || user.login == assignee) continue;
+    if (!review.body_html) {
+      logger.info(`incentivizePullRequestReviews: Skipping to parse the comment because body_html is undefined. comment: ${JSON.stringify(review)}`);
+      continue;
+    }
+    if (!prReviewsByUser[user.login]) {
+      prReviewsByUser[user.login] = [];
+    }
+    prReviewsByUser[user.login].push(review.body_html);
+  }
+  const tokenSymbol = await getTokenSymbol(paymentToken, rpc);
+  logger.info(`incentivizePullRequestReviews: Filtering by the user type done. commentsByUser: ${JSON.stringify(prReviewsByUser)}`);
+
+  // The mapping between gh handle and comment with a permit url
+  const reward: Record<string, string> = {};
+
+  // The mapping between gh handle and amount in ETH
+  const fallbackReward: Record<string, Decimal> = {};
+  let comment = `#### Reviewer Rewards\n`;
+  for (const user of Object.keys(prReviewsByUser)) {
+    const comments = prReviewsByUser[user];
+    const commentsByNode = await parseComments(comments, ItemsToExclude);
+    const rewardValue = calculateRewardValue(commentsByNode, incentives);
+    if (rewardValue.equals(0)) {
+      logger.info(`incentivizePullRequestReviews: Skipping to generate a permit url because the reward value is 0. user: ${user}`);
+      continue;
+    }
+    logger.info(`incentivizePullRequestReviews: Comment parsed for the user: ${user}. comments: ${JSON.stringify(commentsByNode)}, sum: ${rewardValue}`);
+    const account = await getWalletAddress(user);
+    const amountInETH = rewardValue.mul(baseMultiplier);
+    if (amountInETH.gt(paymentPermitMaxPrice)) {
+      logger.info(`incentivizePullRequestReviews: Skipping comment reward for user ${user} because reward is higher than payment permit max price`);
+      continue;
+    }
+    if (account) {
+      const { payoutUrl } = await generatePermit2Signature(account, amountInETH, issue.node_id);
+      comment = `${comment}### [ **${user}: [ CLAIM ${amountInETH} ${tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n`;
```

Comment:
Will this be difficult to consolidate all of the contributors' rewards into a single payment permit?

Reply:
Hmm, of course its possible to consolidate all the rewards among incentive for creatorcomment, commmentsOnIssue, pullrequestReviews.

But does it make more sense to divide them so that contributors can easily know about their rewards in detail.
At least we need to inform them each amount for three incentive types via comment even we consolidate payment permits.

Reply:
hmm, should I stop this pr until https://github.com/ubiquity/ubiquibot/issues/686 is fixed?

Reply:
I would hope that they can somehow be done in tandem.